### PR TITLE
Standardize on ej:valid validation

### PR DESCRIPTION
**Pull request by christophermaier**
This PR makes use of `ej:valid` validation everywhere.  It also removes quite a bit of now-dead code.

These PRs are required for full effect:
- https://github.com/opscode/chef_wm/pull/21
- https://github.com/opscode/chef-pedant-core/pull/10
- https://github.com/opscode/chef-pedant-tests/pull/8

### CODE REVIEW

File: src/chef_data_bag_item.erl [hunk 1]
```diff
@@ -35,11 +34,11 @@
 
 -include("chef_types.hrl").
 
+%% Describes the valid structure of a data bag item for use with `ej:valid/2`.
 -define(VALIDATION_CONSTRAINTS,
-        [
-         {<<"id">>,             {match, "^[[:alnum:]_\:\.\-]+$"}}
-        ]).
-
+        {[
```
Comment (seth) [inline src/chef_data_bag_item.erl:39]: Just a note that this is a case where we very much want to allow any other keys and values.

File: src/chef_data_bag_item.erl [hunk 2]
```diff
@@ -58,39 +57,41 @@ wrap_item(BagName, ItemName, Item) ->
...
+%% @doc For data bag update, the id in URL must match the id (if provided) in the JSON.
+%% Missing the id is ok (you know from the URL you're hitting), but an id mismatch is not.
+%%
+%% Returns `DataBagItem` with `IdFromUrl` unambiguously set as its id, or throws if it was
+%% mismatched.
+-spec normalized_data_bag_for_update(ej:json_object(), binary()) ->ej:json_object().
+normalized_data_bag_for_update(DataBagItem, IdFromUrl) ->
+    case ej:get({<<"id">>}, DataBagItem)of
+        Id when Id =:= IdFromUrl orelse Id =:= undefined ->
+            ej:set({<<"id">>}, DataBagItem, IdFromUrl);
```
Comment (seth) [inline src/chef_data_bag_item.erl]: minor, but looks like we set the name if it already matches. Maybe an extra clause is worthwhile?

File: src/chef_node.erl [hunk 3]
```diff
@@ -31,16 +31,19 @@
...
-         {<<"default">>,          is_ejson_proplist},
-         {<<"override">>,         is_ejson_proplist},
-         {<<"automatic">>,        is_ejson_proplist},
-         {<<"run_list">>,         is_run_list}
-        ]).
+-define(VALIDATION_CONSTRAINTS,
+        {[{<<"name">>, {string_match, chef_regex:regex_for(node_name)}},
+          {<<"chef_environment">>, {string_match, chef_regex:regex_for(environment_name)}},
+          {<<"json_class">>, <<"Chef::Node">>},
+          {<<"chef_type">>, <<"node">>},
```
Comment (seth) [inline src/chef_node.erl:38]: are these required or optional?

Comment (christophermaier) [inline src/chef_node.erl:38]: They are required; we insert default values prior to validating, so there will always be something there.

**Pull request merged**